### PR TITLE
include_in_utilization in model, filter, and editable to support microrequests issue #1084

### PR DESCRIPTION
**Pull request by cryptofilegsa**
## Description

This pull request provides underlying database and admin filtering to support project level overrides to both include or exclude from utilization reporting.  Reference issue:  https://github.com/18F/tock/issues/1084

For each Project/Tock line:

Added an include_in_utilization boolean field, defaults to that project's billable status

include_in_utilization defaults to False but can be custom-set to True in the admin interface.

(can be set for the 4 edge cases we run into every week)

Further work:

Look into how utilization reports could filter on the new Project.include_in_utilization field

## Additional information

yellow highlighting of the enhancements in the projects admin page:

* Screenshot images
image

### CODE REVIEW

File: tock/projects/models.py [hunk 1]
```diff
@@ -247,6 +247,11 @@ class Project(models.Model):
         default=False,
         help_text='Check if this project should be excluded from calculations of billable hours, e.g. Out of Office'
     )
+    include_in_utilization = models.BooleanField(
+            default=False,
```
Comment (Jkrzy) [inline tock/projects/models.py:251]: For all existing Projects, we want to set the value of this to the project's billable status. This sets _all_ of them to `False`

File: tock/projects/models.py [hunk 2]
```diff
@@ -247,6 +247,11 @@ class Project(models.Model):
...
+            help_text="For handling microrequests, per github.com/18F/tock/issues/1084"
```
Comment (Jkrzy) [inline tock/projects/models.py:252]: There's an opportunity here to use the `help_text` attribute to describe the functionality that setting this field to `True` provides. Noting the origin in the code does provide much utility.

File: tock/hours/admin.py [hunk 3]
```diff
@@ -86,7 +87,7 @@ class TimecardAdmin(admin.ModelAdmin):
     inlines = (TimecardObjectInline,)
     list_display = ('user', 'reporting_period', 'billable_expectation', 'organization', 'submitted')
     list_editable = ('organization', 'billable_expectation')
-    list_filter = (ReportingPeriodListFilter, 'reporting_period',)
+    list_filter = ( 'user', ReportingPeriodListFilter, 'reporting_period')
```
Comment (Jkrzy) [inline tock/hours/admin.py:90]: I believe these changes are already in `main`

File: tock/projects/models.py (hunk 1 above)
Comment (cryptofilegsa) [reply tock/projects/models.py]: tactically, starting with the most common value and can update in separate PR

File: tock/projects/models.py (hunk 2 above)
Comment (cryptofilegsa) [reply tock/projects/models.py:254]: addressed in a9259f2

File: tock/projects/models.py (hunk 1 above)
Comment (timoballard) [reply tock/projects/models.py:253]: I think there are a couple of different ways to handle this, but I tend to extend the auto-generated migration to set the defaults. Here's an example:

https://github.com/18F/tock/blob/1b8b88039d4b52e8c184281e392c1fde65c890c6/tock/hours/migrations/0055_auto_20200430_1742.py#L7-L40

**Comment by timoballard** [comment]
Left a comment about setting the `include_in_utilization` default value based on billable status as part of the migration. Happy to pair on this if you'd like, @cryptofilegsa

File: tock/projects/migrations/0027_project_include_in_utilization.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Generated by Django 2.2.13 on 2020-06-23 16:29
+
+from django.db import migrations, models
+
+def make_default_utilization(apps, schema_editor):
+    Projects = apps.get_model('projects','Project')
+
+    for project in Projects.objects.all():
+        project.include_in_utilization = project.is_billable
```
Comment (timoballard) [inline tock/projects/migrations/0027_project_include_in_utilization.py:9]: turns out `is_billable` isn't accessible in this context - if you switch this to...

`project.include_in_utilization = project.accounting_code.billable`

...you should be good.

**Comment by tbaxter-18f** [comment]
👋  would this be useful to finish up?

**Comment by Jkrzy** [comment]
@tbaxter-18f, if you're still up for it -- absolutely this would be good to wrap up!

**Comment by kfoley-18F** [comment]
No longer needed with weekly billing.

**Pull request closed without merging**